federatedscope/core/auxiliaries/trainer_builder.py

```diff
@@ -108,6 +108,11 @@ def get_trainer(model=None,
             raise ValueError('Trainer {} is not provided'.format(
                 config.trainer.type))
 
+    if not isinstance(trainer, Trainer):
```

Comment:
It might be hard for users to interpret this warning message. At least I need a second-thought